### ISSUE TITLE
Error in file(con, "r") : cannot open the connection

### ISSUE THREAD

**Issue by BranBlack3125** [nwfsc-assess/geostatistical_delta-GLMM#9]
Hello everyone,

The following is a problem I've encountered while attempting to run the Example-simple script posted for geostatistical_delta-GLMM. Below I've copied a post on this issue that I originally submitted to StackOverflow:

I am a complete R novice, and have been attempting to run the following package/model in RStudio 3.2.4 (https://github.com/nwfsc-assess/geostatistical_delta-GLMM). The specific script I am attempting to run can be found here: (https://github.com/nwfsc-assess/geostatistical_delta-GLMM/blob/master/examples/Example--simple.R). The script (second link) is an example code posted online to help people learn how to run this model (first link). It references several datasets, which I have saved in the location indicated by the TmbDIR variable in my example code below. I have been getting the following error once the script gets to the "Make and Run TMB model" section" (sorry, I don't see any labels for line number - this error occurs about 3/4 of the way through the script):

Make TMB object

TmbDir = "C:/Bran/HUD/nwsfc_assess_geostatistical_delta_GLMM/geostatistical_delta-GLMM-master/data/" TmbList = Build_TMB_Fn("TmbData"=TmbData, "TmbDir"=TmbDir, "RunDir"=DateFile, "Version"=Version, "RhoConfig"=RhoConfig, "VesselConfig"=VesselConfig, "loc_x"=Spatial_List$loc_x) Error in file(con, "r") : cannot open the connection In addition: Warning message: In file(con, "r") : cannot open file 'geo_index_v4a.cpp': No such file or directory
From there on out in that section of the script, I get multiple errors warning that various objects the script is supposed to be accessing have not been found.

I have attempted the following for troubleshooting: I installed RTools, and reinstalled INLA and TMB (the second to are called on by the script). From my Google search on the error I also saw that many people with this error have used "setInternet2(TRUE)" and solved the problem, but that didn't work in my case.

Unfortunately, as I mentioned earlier, I am a complete novice at R and don't have any good ideas on what to try next - I imagine it's quite likely this error is due to simple user error that I'm simply not experienced enough to spot.

Thank you!

**Comment by James-Thorson** [comment]
Please try again running the slightly modified version of
Example-Simple.R.  The previous version was pointing at the location of the
TMB CPP file on my machine, rather than the general version.  I also added
an informative warning about this problem to Build_TMB_Fn

jim

**Comment by BranBlack3125** [comment]
Thank you for posting the new Example-Sample.R so quickly. I'm sorry, but I
tried running it and ended up with a different error in the same place:

> # Make TMB object
> TmbList = Build_TMB_Fn("TmbData"=TmbData, "RunDir"=DateFile,
"Version"=Version, "RhoConfig"=RhoConfig, "VesselConfig"=VesselConfig,
"loc_x"=Spatial_List$loc_x)
Error in compile(paste0(Version, ".cpp")) : Compilation failed
In addition: Warning message:
running command 'make -f "C:/PROGRA~1/R/R-32~1.4RE/etc/x64/Makeconf" -f
"C:/PROGRA~1/R/R-32~1.4RE/share/make/winshlib.mk" -f
"C:\Users\cglab\AppData\Local\Temp\RtmpcxzgUc\file2f68342e7c2f"
SHLIB_LDFLAGS='$(SHLIB_CXXLDFLAGS)' SHLIB_LD='$(SHLIB_CXXLD)'
SHLIB="geo_index_v4a.dll" WIN=64 TCLBIN=64 OBJECTS="geo_index_v4a.o"' had
status 127

On a couple of the runs I tried earlier this week I tried to change the
paths to my machine, but I don't think I did it correctly - I still got the
same error I reported online.

Maybe I just did something weird when I installed everything on my work
computer? I was going to try installing RStudio, RTools, TMB, and INLA on
my personal laptop this weekend and see if I can do a better job the second
time around. I noticed that it looked like you uploaded your scripts when
RStudio 3.2.4 was the newest release, so I installed 3.2.4. Is that the
version you would suggest, or is that not important?

Thank you.

- Bran

**Comment by James-Thorson** [comment]
Yes, that's an installation issue with TMB rather than a problem with my
software per se.  Please see the installation instructions here:

https://github.com/James-Thorson/2016_Spatio-temporal_models

and please confirm that you can run a simple TMB example.  i'll brainstorm
more tomorrow during a meeting

jim

**Comment by BranBlack3125** [comment]
Ok, that makes sense. I thought it was pretty likely I may have goofed
something up with the installation. I will try a new install on my laptop
following the link you provided this weekend and let you know how it turns
out.

- Bran
